### PR TITLE
Duplicate a course, but without the full content.

### PR DESCRIPTION
**Pull request by fonglh**
References #431.

This PR allows a course to be duplicated, but without any of the content. The duplicated course will have the same assessment categories and tabs, and will be time shifted by a user selected amount of time.

I'll prefer to merge this in piece by piece instead of a huge merge when duplication for the entire course is ready.

Commit 5193e5a should be removed when everything else is ready.

Don't merge yet, assessment categories and tabs may have to be time shifted.

### CODE REVIEW

**Comment by weiqingtoh** [comment]
`assessment categories and tabs may have to be time shifted` -> do you mean the assessments in the tabs need to be timeshifted? 

Not sure if we should add all the initialize_duplicator into a separate concern for each model - thoughts?

**Comment by fonglh** [comment]
No, it's the `material` folder that has start/end times. I noticed something else in the db that wasn't time shifted but couldn't remember exactly what it was when writing the PR description.

File: app/services/course/duplication_service.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  def duplicator
+    @duplicator ||= Duplicator.new(@all_objects - @selected_objects, time_shift)
+  end
+
+  # Calculate the amount of time the objects in the new course have to be shifted by
+  #
+  # @return [ActiveSupport::TimeWithZone]
+  def time_shift
+    # must be symbol key
+    @duplication_params[:new_course_start_date] - @current_course.start_at
```
Comment (weiqingtoh) [inline app/services/course/duplication_service.rb]: Perhaps you can cater for a missing `:new_course_start_date` params. In that case, a reasonable default will do, so time-shift will then be `0`? (Or maybe some time after `Time.zone.now`?)

Comment (fonglh) [inline app/services/course/duplication_service.rb]: I think a reasonable default will be from the day the course is duplicated.

The form enforces the entry of `new_course_start_date`, so unless someone purposely tries to bypass it, it shouldn't happen.

**Comment by fonglh** [comment]
Will close this and reopen as a full feature when an entire course can be duplicated.

**Pull request closed without merging**